### ISSUE TITLE
Extra empty line added before comments with // in method body

### ISSUE THREAD

**Issue by murdos** [jhipster/prettier-java#270]
Is this behavior expected?

Before (from JHipster WebConfigurer.java)
```
    private void setMimeMappings(WebServerFactory server) {
        if (server instanceof ConfigurableServletWebServerFactory) {
            MimeMappings mappings = new MimeMappings(MimeMappings.DEFAULT);
            // IE issue, see https://github.com/jhipster/generator-jhipster/pull/711
            mappings.add("html", MediaType.TEXT_HTML_VALUE + ";charset=" + StandardCharsets.UTF_8.name().toLowerCase());
            // CloudFoundry issue, see https://github.com/cloudfoundry/gorouter/issues/64
            mappings.add("json", MediaType.TEXT_HTML_VALUE + ";charset=" + StandardCharsets.UTF_8.name().toLowerCase());
            ConfigurableServletWebServerFactory servletWebServer = (ConfigurableServletWebServerFactory) server;
            servletWebServer.setMimeMappings(mappings);
        }
    }
```

Actual:
```
    private void setMimeMappings(WebServerFactory server) {
        if (server instanceof ConfigurableServletWebServerFactory) {
            MimeMappings mappings = new MimeMappings(MimeMappings.DEFAULT);

            // IE issue, see https://github.com/jhipster/generator-jhipster/pull/711
            mappings.add("html", MediaType.TEXT_HTML_VALUE + ";charset=" + StandardCharsets.UTF_8.name().toLowerCase());

            // CloudFoundry issue, see https://github.com/cloudfoundry/gorouter/issues/64
            mappings.add("json", MediaType.TEXT_HTML_VALUE + ";charset=" + StandardCharsets.UTF_8.name().toLowerCase());
            ConfigurableServletWebServerFactory servletWebServer = (ConfigurableServletWebServerFactory) server;
            servletWebServer.setMimeMappings(mappings);
        }
    }
```

**Comment by murdos** [comment]
A side effect is that a new line is added before needles in code generated by JHipster
```

// jhipster-needle-entity-add-getters-setters - JHipster will add getters and setters here, do not remove
```

**Comment by clement26695** [comment]
It is not done on purpose, but it is difficult to resolve this issue right now. As we might have to refactor the way we handle comments to fix #262, it might help to resolve this as well

**Comment by pascalgrimaud** [comment]
Adding a bounty on this, as it's a bug and it will help the project
